inc/class-tmsc.php

```diff
@@ -1,106 +1,314 @@
+	 * Add the Zoninator taxonomy to the custom post types that require it.
+	 * The function calls below just return false if either the taxonomy or post type doesn't exist.
+	 *
+	 * @access public
+	 * @return void
+	 */
+	public function zoninator_post_type_support() {
+
+		$post_types = array( 'tms_object' );
 
-			// Add zoninator post type support
-			add_action( 'zoninator_post_init', array( $this, 'zoninator_post_type_support' ), 99 );
-			add_filter( 'zoninator_recent_posts_args', array( $this, 'filter_zone_post_types' ) );
-			add_filter( 'zoninator_search_args', array( $this, 'filter_zone_post_types' ) );
+		// Add the Zoninator taxonomy for the defined post types
+		foreach ( $post_types as $post_type ) {
+			add_post_type_support( $post_type, 'zoninator_zones' );
+			register_taxonomy_for_object_type( 'zoninator_zones', $post_type );
 		}
 
-		/**
-		 * Add the Zoninator taxonomy to the custom post types that require it.
-		 * The function calls below just return false if either the taxonomy or post type doesn't exist.
-		 *
-		 * @access public
-		 * @return void
-		 */
-		public function zoninator_post_type_support() {
+		// The Zoninator internal post_types array has already been written and cached by this point.
+		// We must override it to enabled these post types for the recent posts dropdown and AJAX searches.
+		z_get_zoninator()->post_types = array_merge( z_get_zoninator()->post_types, $post_types );
+	}
 
-			$post_types = array( 'tms_object' );
+	/**
+	 * Only display tmsc objects in our TMSConnect Zones.
+	 */
+	public function filter_zone_post_types( $args ) {
+		$active_zone_id = z_get_zoninator()->_get_request_var( 'zone_id', 0, 'absint' );
+		$tmsc_zones = wp_list_pluck( $this->default_zones, 'slug' );
+		if ( ! empty( $active_zone_id ) ) {
+			$zone = z_get_zone( $active_zone_id );
+			if ( in_array( $zone->slug, $tmsc_zones ) ) {
+				$args['post_type'] = array( 'tms_object' );
+			}
+		}
+
+		return $args;
+	}
 
-			// Add the Zoninator taxonomy for the defined post types
-			foreach ( $post_types as $post_type ) {
-				add_post_type_support( $post_type, 'zoninator_zones' );
-				register_taxonomy_for_object_type( 'zoninator_zones', $post_type );
+	/**
+	 * Generate our sync options admin area.
+	 *
+	 */
+	public function sync_init() {
+		$fm = new \Fieldmanager_Group( array(
+			'name' => 'tmsc_guide_terms',
+			'children' => array(
+				'term' => new \Fieldmanager_Group( array(
+					'description' => __( 'Use the CN number of the guide terms that should serve as parent taxonomies.', 'tmsc' ),
+					'collapsible' => true,
+					'label' => __( 'Guide Terms', 'tmsc' ),
+					'children' => array(
+						'data' => new \Fieldmanager_Group( array(
+							'limit' => 0,
+							'add_more_label' => __( 'Add another guide term', 'tmsc' ),
+							'children' => array(
+								'taxonomy_map' => new \Fieldmanager_Select( __( 'Map to Taxonomy', 'tmsc' ), array(
+									'datasource' => new \Fieldmanager_Datasource( array(
```

Comment:
Was `Fieldmanager_Datasource_Term` not working for you here? Just wondering since it seems like you can just use that instead.